File: Backend/Database/Models/Delivery.cs
using System.ComponentModel.DataAnnotations;

namespace Database.Models;

public class Delivery
{
    [Key]
    public required Guid Id { get; set; }
    public required Guid AssignmentId { get; set; }
    public Assignment? Assignment { get; set; }
    public List<DeliveryField>? Fields { get; set; }
    public required Guid? StudentId { get; set; }
    public User? Student { get; set; }
    public required Guid? TeamId { get; set; }
    public Team? Team { get; set; }
}

File: Frontend/src/types/delivery.ts
export type DeliveryField = {
    id: string;
    assignmentFieldId: string;
    value: any;
}

export type Delivery = {
    id: string;
    assignmentId: string;
    studentId?: string;
    teamId: string;
    fields: DeliveryField[];
}

export type CreateDelivery = {
    assignmentId: string;
    studentId: string;
    fields: CreateDeliveryField[];
}

export type UpdateDelivery = {
    fields: CreateDeliveryField[];
}

export type CreateDeliveryField = {
    assignmentFieldId: string;
    value: any;
}